deepmd/train/trainer.py

```diff
@@ -470,12 +211,9 @@ def build(self, data=None, stop_batch=0, origin_type_map=None, suffix=""):
 
         if not self.multi_task_mode:
             if not self.is_compress and data.mixed_type:
-                assert self.descrpt_type in [
-                    "se_atten"
-                ], "Data in mixed_type format must use attention descriptor!"
-                assert self.fitting_type in [
-                    "ener"
-                ], "Data in mixed_type format must use ener fitting!"
+                assert isinstance(
+                    self.fitting, EnerFitting
+                ), "Data in mixed_type format must use ener fitting!"
```

Comment:
Why delete 'se_atten' descriptor judgement for mixed_type?

Reply:
This PR uncouples the descriptor and the fitting from the trainer, as a trainer may not contain only one descriptor.
Theoretically, we should check it directly in `descriptor.build`. However, I think #2481 is a better idea to judge `mixed_type`. We don't need it if #2481 is merged.